pkg/queue/cluster_queue_impl.go

```diff
@@ -66,30 +71,16 @@ func (c *ClusterQueueImpl) DeleteFromQueue(q *Queue) {
 // pushIfNotPresent pushes the workload to ClusterQueue.
 // If the workload is already present, returns false. Otherwise returns true.
 func (c *ClusterQueueImpl) pushIfNotPresent(info *workload.Info) bool {
-	item := c.heap.items[workload.Key(info.Obj)]
-	if item != nil {
-		return false
-	}
-	heap.Push(&c.heap, *info)
-	return true
+	return c.heap.AddIfNotPresent(info)
 }
 
 func (c *ClusterQueueImpl) PushOrUpdate(w *kueue.Workload) {
-	item := c.heap.items[workload.Key(w)]
-	info := *workload.NewInfo(w)
-	if item == nil {
-		heap.Push(&c.heap, info)
-		return
-	}
-	item.obj = info
-	heap.Fix(&c.heap, item.index)
+	info := workload.NewInfo(w)
+	c.heap.AddOrUpdate(info)
 }
 
 func (c *ClusterQueueImpl) Delete(w *kueue.Workload) {
-	item := c.heap.items[workload.Key(w)]
-	if item != nil {
-		heap.Remove(&c.heap, item.index)
-	}
+	_ = c.heap.Delete(workload.Key(w))
```

Comment:
whenever you ignore an error, explain why. But I think we should remove the return value.

Reply:
changed as advised.